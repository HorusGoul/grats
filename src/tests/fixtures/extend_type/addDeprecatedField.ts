/** @GQLType */
class Query {
  // No fields
}

/**
 * @GQLExtendType
 * @deprecated Because reasons
 */
export function greeting(query: Query): string {
  return "Hello world!";
}
