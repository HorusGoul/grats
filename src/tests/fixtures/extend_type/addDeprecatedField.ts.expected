-----------------
INPUT
----------------- 
/** @gqlType */
class SomeType {
  // No fields
}

/**
 * @gqlField
 * @deprecated Because reasons
 */
export function greeting(query: SomeType): string {
  return "Hello world!";
}

-----------------
OUTPUT
-----------------
-- SDL --
directive @asyncIterable on FIELD_DEFINITION

directive @propertyName(name: String!) on FIELD_DEFINITION

directive @exported(tsModulePath: String!, functionName: String!, argCount: Int!) on FIELD_DEFINITION

directive @exportedScalar(tsModulePath: String!, exportName: String!) on SCALAR

type SomeType {
  greeting: String @deprecated(reason: "Because reasons") @exported(tsModulePath: "grats/src/tests/fixtures/extend_type/addDeprecatedField.ts", functionName: "greeting", argCount: 1)
}
-- TypeScript --
import { greeting as someTypeGreetingResolver } from "./addDeprecatedField";
import { GraphQLSchema, GraphQLObjectType, GraphQLString } from "graphql";
function getSchema(): GraphQLSchema {
    const SomeTypeType: GraphQLObjectType = new GraphQLObjectType({
        name: "SomeType",
        fields() {
            return {
                greeting: {
                    deprecationReason: "Because reasons",
                    name: "greeting",
                    type: GraphQLString,
                    resolve(source) {
                        return someTypeGreetingResolver(source);
                    }
                }
            };
        }
    });
    return new GraphQLSchema({
        types: [SomeTypeType]
    });
}
export { getSchema };
